Fix LIST and UIDL with msg-nbr, added LAST command

metafloor opened the pull request:
My testing found that the LIST and UIDL commands hang when a message number is supplied.  This was caused by a multi-line stream always being created.  When a message number is present, the data is only in the `info` response and no stream is needed.

I also added the LAST command as it is useful for stateful clients.

brettz9 commented:
Thanks for this. Could you add tests please? You can set up the testing environment by adding your own `pop.config.json` file at the project root (see pop.config-sample.json for the potential values) to point to a mail server testing account. A few existing tests currently have or sometimes have issues (perhaps including one you now fixed).

metafloor commented:
Looking at the `test/programmatic.mjs`, there are `skip()s` on the LIST/UIDL test cases that use a message number.  Those test cases are exactly what this PR fixes.  I have updated the file to remove the skip()s and added a LAST test case.  New commit has been added to the PR.

brettz9 commented:
Could you rebase onto the latest on `master` and let me know how the tests go?

brettz9 commented:
I've incorporated into `master` (and also started seeding messages in the tests and auto-deleting them). Thanks for your contributions!
